src/carma-streets/plot_message_frequencies.py

```diff
@@ -0,0 +1,103 @@
+                message_data_frame["Time (s)"] = message_data_frame["Time (ms)"]/1000
+            else :
+                message_data_frame["Time (s)"] = message_data_frame["Timestamp(ms)"]/1000
+            message_data_frame = get_message_frequency(message_data_frame)
+            if KafkaLogMessageType.MAP.value in message_name:
+                #
+                plot_message_frequency(plots[idx],message_data_frame['Time (s)'], message_data_frame["Average Frequency (Hz)"] ,message_name,1, 1)
+            else:
+                # Any message with 10 Hz as target frequency
+                plot_message_frequency(plots[idx],message_data_frame['Time (s)'], message_data_frame["Average Frequency (Hz)"],message_name)
+            fig.suptitle('Message Frequency Plots')
+            fig.supxlabel('Time (s)')
+            fig.supylabel('Message Frequency (Hz)')
+            fig.savefig(f'{plots_dir}/message_frequencies.png')
+        
+def plot_message_frequency( axes: axes.Axes, time: list, frequency: list , message_name: str, target_frequency: int = 10, absolute_error: int = 2) :
+    """Generate subplots for each message frequency
+
+    Args:
+        axes (axes.Axes): Subplot of message frequency
+        time (list): Time data (s)
+        frequency (list): Frequency data (Hz)
+        message_name (str): Message Name
+        target_frequency (int, optional): Target frequency. Defaults to 10.
+        absolute_error (int, optional): Acceptable deviation from target frequency. Defaults to 2.
+    """
+    axes.scatter(time,frequency, c="blue", marker="^", label="Freq (hz)")
+    # Add horizontal lines for differing freq requirements based on message type
+    axes.axhline(y=target_frequency-absolute_error, color='r', linestyle='--', label="frequency lower bound")
+    axes.axhline(y=target_frequency+absolute_error, color='r', linestyle='-', label="frequency upper bound")
+    axes.set_title(message_name)
+    axes.set_ylim(target_frequency-2*absolute_error, target_frequency+2*absolute_error)
+
+def get_simulation_time(message_wall_time : list, time_sync_wall_time: list, time_sync_simulation_time : list)-> list:
+    message_simulation_time=list()
+    for msg_wall_time in message_wall_time:
+        for idx, t_sync_wall_time in enumerate(time_sync_wall_time):
+            if idx == 0:
+                continue
+            elif msg_wall_time >= time_sync_wall_time[idx-1] and msg_wall_time < time_sync_wall_time[idx]:
+                message_simulation_time.append(time_sync_simulation_time[idx-1])
+                break
+            elif msg_wall_time >= time_sync_wall_time[idx] and idx == len(time_sync_wall_time)-1:
+                message_simulation_time.append(time_sync_simulation_time[idx])
+                break
+    return message_simulation_time
+def get_message_frequency( messages: pd.DataFrame) -> pd.DataFrame:
```

Comment:
The function name is a bit misleading. The `get_` prefix implies that it will not modify `messages`, but the implementation adds two columns to the `DataFrame`. Consider renaming it `add_frequency_columns()` or something similar and changing the return type to `None`. For example:

```python
def add_frequency_columns(messages: pd.DataFrame):
    messages["Instantaneous Frequency (Hz)"] = 1 / messages["Time (s)"].diff() 
    messages["Average Frequency (Hz)"] = messages["Instantaneous Frequency (Hz)"].rolling(window=50, min_periods=1).mean()

# ...

add_frequency_columns(messages)
```